src/Settings.pm

```diff
@@ -428,14 +428,26 @@ sub loadByHandle {
 	assert(defined $object) if DEBUG;
 
 	my $filename;
-	if ($object->{autoSearch}) {
-		if ($object->{type} == CONTROL_FILE_TYPE) {
-			$filename = _findFileFromFolders($object->{name}, \@controlFolders);
+	my $internalFilename;
+	if ($object->{'autoSearch'}) {
+		$internalFilename = $object->{'name'};
+	} else {
+		$internalFilename = $object->{'internalName'};
+	}
+
+	#hooks of type 'pre_load_' make it possible to change the filename before openkore searches for it (this filename must contain file name and file path)
+	my %pre_load = (internalFilename => $internalFilename, filename => \$filename);
+	Plugins::callHook('pre_load_'.$internalFilename, \%pre_load);
```

Comment:
I'm on a crusade against deeply nested logic. :)

How about:
```
my $pre_load = {internalFilename => $internalFilename, filename => $filename};
Plugins::callHook('pre_load_'.$internalFilename, $pre_load);
if ($pre_load->{return}) {
	$filename = $pre_load->{filename};
} elsif ($object->{autoSearch} && $object->{type} == CONTROL_FILE_TYPE) {
	$filename = _findFileFromFolders($object->{name}, \@controlFolders);
} elsif ($object->{autoSearch}) {
	$filename = _findFileFromFolders($object->{name}, \@tablesFolders);
} else {
	$filename = $object->{name};
}
```

Also, should we pass the `$object->{type}` as well, so plugins can make the same sort of decisions that this code does?